assimilation_code/modules/assimilation/adaptive_inflate_mod.f90

```diff
@@ -757,60 +739,60 @@ subroutine bayes_cov_inflate(ens_size, inf_type, x_p, sigma_p_2, y_o, sigma_o_2,
+                                       rate, gamma_corr, new_cov_inflate)
    
-      ! Computational errors check (small numbers + NaNs)
-      if (abs(density_1) <= TINY(0.0_r8) .OR. &
-          abs(density_2) <= TINY(0.0_r8) .OR. &
-          density_1 /= density_1 .OR. density_1 /= density_1 .OR. &
-          density_2 /= density_2 .OR. density_2 /= density_2) then
-         new_cov_inflate_sd = lambda_sd
-         return
-      endif
+   ! Computational errors check (small numbers + NaNs)
+   if (abs(density_1) <= TINY(0.0_r8) .OR. &
+       abs(density_2) <= TINY(0.0_r8) .OR. &
+       density_1 /= density_1 .OR. density_1 /= density_1 .OR. &
+       density_2 /= density_2 .OR. density_2 /= density_2) then
+      new_cov_inflate_sd = lambda_sd
+      return
+   endif
    
-      ! Now, compute omega and the new distribution parameters
-      ratio     = density_1 / density_2
-      omega     = log(new_cov_inflate          )/new_cov_inflate + 1.0_r8/new_cov_inflate - &
-                  log(new_cov_inflate+lambda_sd)/new_cov_inflate - 1.0_r8/(new_cov_inflate+lambda_sd)
-      rate_new  = log(ratio) / omega
-      shape_new = rate_new / new_cov_inflate - 1.0_r8
+   ! Now, compute omega and the new distribution parameters
+   ratio     = density_1 / density_2
+   omega     = log(new_cov_inflate          )/new_cov_inflate + 1.0_r8/new_cov_inflate - &
+               log(new_cov_inflate+lambda_sd)/new_cov_inflate - 1.0_r8/(new_cov_inflate+lambda_sd)
+   rate_new  = log(ratio) / omega
+   shape_new = rate_new / new_cov_inflate - 1.0_r8
    
-      ! Finally, get the sd of the IG posterior
-      if (shape_new <= 2.0_r8) then
-         new_cov_inflate_sd = lambda_sd
-         return
-      endif
-      new_cov_inflate_sd = sqrt(rate_new**2 / ( (shape_new-1.0_r8)**2 * (shape_new-2.0_r8) ))
+   ! Finally, get the sd of the IG posterior
+   if (shape_new <= 2.0_r8) then
+      new_cov_inflate_sd = lambda_sd
+      return
+   endif
+   new_cov_inflate_sd = sqrt(rate_new**2 / ( (shape_new-1.0_r8)**2 * (shape_new-2.0_r8) ))
    
-      ! If the updated variance is more than xx% the prior variance, keep the prior unchanged 
-      ! for stability reasons. Also, if the updated variance is NaN (not sure why this
-      ! can happen; never did when developing this code), keep the prior variance unchanged. 
-      if ( new_cov_inflate_sd > sd_max_change_in*lambda_sd .OR. &
-           new_cov_inflate_sd /= new_cov_inflate_sd) then
-         new_cov_inflate_sd = lambda_sd
-         return
-      endif
+   ! If the updated variance is more than xx% the prior variance, keep the prior unchanged 
+   ! for stability reasons. Also, if the updated variance is NaN (not sure why this
+   ! can happen; never did when developing this code), keep the prior variance unchanged. 
+   if ( new_cov_inflate_sd > sd_max_change_in*lambda_sd .OR. &
+        new_cov_inflate_sd /= new_cov_inflate_sd) then
+      new_cov_inflate_sd = lambda_sd
+      return
    endif
    
 else
```

Comment:
does this else still match the same if() now that you've removed an else block?  perhaps adding a comment at the end of this line with a hint to the if() context would help.